Source/Core/CredentialsRegistry.js

```diff
@@ -0,0 +1,135 @@
+/*global define*/
+define([
+    './defined',
+    './defineProperties',
+    './DeveloperError',
+    '../ThirdParty/Uri'
+], function(
+    defined,
+    defineProperties,
+    DeveloperError,
+    Uri) {
+    'use strict';
+    
+    /**
+     * A singleton that contains all of the servers that are trusted. Credentials will be sent with
+     * any requests to these servers.
+     *
+     * @exports CredentialsRegistry
+     *
+     * @see {@link http://www.w3.org/TR/cors/|Cross-Origin Resource Sharing}
+     */
+    var CredentialsRegistry = {};
+    var _credentials = {};
+
+    /**
+     * Adds a trusted server to the registry
+     *
+     * @param {String} authority The server that is trusted. Can include a port if it is needed. (eg. my.server.com or my.server.com:8080)
+     *
+     * @example
+     * // Add a trusted server
+     * CredentialsRegistry.addTrustedServer('my.server.com:81');
+     */
+    CredentialsRegistry.addTrustedServer = function(authority) {
```

Comment:
Should this be modeled after standard Cesium collections, e.g., `TrustedServers.add`, `remove`, `contains`.  More consistent and precise, and less verbose.  Or is `CredentialsRegistry` a standard term and an object that will expand?